tests/integration/ui_write/base.py

```diff
@@ -188,3 +191,28 @@ def addContainer(self, nwbfile):
     def getContainer(self, nwbfile):
         ''' Should take an NWBFile object and return the Container'''
         return nwbfile.get_acquisition(self.container.name)
+
+
+class TestMapRoundTripWithExistingFileObject(TestMapRoundTrip):
```

Comment:
Can't you just implement setUpContainer, addContainer, getContainer?

Reply:
I appreciate the desire to reuse code, but this has made things much more complicated than necessary. There's a lot of extra code and logic in test_roundtrip solely for testing a single feature. I prefer that we keep the general methods upon which the entire integration test suite depends as simple as possible. 

Is there a reason you didn't just duplicate test, and modify it to pass in an h5py.File?

https://github.com/NeurodataWithoutBorders/pynwb/blob/5bcdc8316d81d099514a3df95ad1d884a2c49fd6/tests/integration/test_io.py#L65